File: C/4.cpp
#include <iostream>
using namespace std;

int sum_digits(int n)
{
	int s = 0;
	while(n!=0)
	{
		s += n % 10;
		n /= 10;
	}
	return s;
}

int main()
{
	int m, n, c=0;
	cin >> m;
	cin >> n;
	for(int i = m ; i <=n ; i++)
		if(i % sum_digits(i) == 0)
			c++;
	cout << c;
}

File: python/4.py
#!/usr/bin/env python
# coding: utf-8

# In[ ]:


def sum_digits(n):
    s = 0
    while n!=0:
        s += n % 10
        n //= 10
    return s

c = 0
m = int(input())
n = int(input())
for i in range (m, n+1):
    if i % sum_digits(i) == 0:
        c += 1
print(c)

